### PR TITLE
feat(tab-links): Refactor into new TabLinks component [WD-6842]

### PR DESCRIPTION
**Pull request by aaryanporwal**
## Done

- Refactored all tab usages into a new TabLinks component to reduce code duplication across components.
- Made tabs links, that can be opened into a new tab etc.

Fixes [list issues/bugs if needed]

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Go to storage page, instance, network, profile, storagepool, storagevolume detail pages and check if the tabs work well.
    - Check that the tabs can be right clicked and opened into a new tab etc. (because the tabs now have a href)

## Screenshots

Screen Recording showing tab links

### CODE REVIEW

File: src/components/TabLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useNavigate } from "react-router-dom";
+import { slugify } from "util/slugify";
+
+interface Props {
+  TABS: string[];
+  activeTab?: string;
+  tabUrl: string;
+}
+
+const TabLinks: FC<Props> = ({ TABS, activeTab, tabUrl }) => {
```
Comment (edlerd) [inline src/components/TabLinks.tsx:13]: The prop naming can be improved, how about `tabs, activeTab, baseUrl`? Prop names should always be `camlCase`.

File: src/components/TabLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        return {
+          label: tab,
+          id: tabPath,
+          active: tabPath === activeTab || (tab === TABS[0] && !activeTab),
+          onClick: (e) => {
+            e.preventDefault();
+            handleTabChange(tabUrl, tabPath);
+          },
+          href: tabPath === TABS[0] ? tabUrl : `${tabUrl}/${tabPath}`,
```
Comment (edlerd) [inline src/components/TabLinks.tsx:41]: Can we put this calculation into line 32, something like:
```
const href=tabPath === TABS[0] ? tabUrl : `${tabUrl}/${tabPath}`
```

Then we can use `href` directly.
And we can remove the `handleTabChange` function. Move the two lines `navigate(href)` and `notify.clear()` into the `onClick` handler.
This avoids the duplicate calculation of the href and should simplify the component a little bit.

**Review by edlerd — COMMENTED** [review]
Looks excellent, thanks :)
Some ideas to simplify below.

File: src/components/TabLinks.tsx (hunk 2 above)
Comment (aaryanporwal) [reply src/components/TabLinks.tsx:41]: Good idea 👍

**Review by edlerd — APPROVED** [review]
Very good, thank you for the changes :+1:

**Pull request merged**